### ISSUE TITLE
Update to PHP 7

### ISSUE THREAD

**Issue by nikic** [phpinternalsbook/PHP-Internals-Book#9]
Right now the book documents PHP 5. Presumably, we'll want to update it to PHP 7. How should we go about this?

I don't think that we should do this update in-place. Instead we should create separate PHP 5 and PHP 7 sections and add content to the PHP 7 section as it is updated.

**Comment by SammyK** [comment]
+1 for separate sections.

**Comment by jpauli** [comment]
Ok I'll start some work soon about this

**Comment by tvlooy** [comment]
What is this book trying to accomplish / who it is for? I am trying to understand how usefull / relevant is it to keep info about 5.

**Comment by jpauli** [comment]
I think atm we should keep PHP 5 as-is, and start a new 'fork' work and focus on PHP 7

**Comment by tvlooy** [comment]
+1 for that

**Comment by nikic** [comment]
To preserve existing URLs we probably can't move the PHP 5 content, so I'd suggest placing all the PHP 7 stuff in a `php7/` directory. In the index.rst TOC we can just create two sections for PHP 5 and PHP 7.

**Comment by beberlei** [comment]
Content for both versions is interesting, but i wouldn't care much about the old urls still pointing to php5. In place modification of the pages is easier for you and as a regular reader probably fine for me, as long as you keep the php 5 version archived somewhere at php5.phpinternalsbook.com and maybe add a link to the php 7 version to go to the old version.

**Comment by Fleshgrinder** [comment]
I agree with @beberlei here, no need to preserve the content under those URLs. Time moves on, Google will keep up. 😉

**Comment by dguhl** [comment]
As this is a book, why not forking it and bring different editions for 5 and 7?
